XenModel/Actions/Network/NetworkingActionHelpers.cs

```diff
@@ -269,6 +269,45 @@ private static void ReconfigureManagement_(AsyncAction action, PIF pif, int hi)
             action.Description = string.Format(Messages.ACTION_CHANGE_NETWORKING_MANAGEMENT_RECONFIGURED, pif.Name);
         }
 
+        internal static void PoolReconfigureManagement(AsyncAction action, PIF up_pif, PIF down_pif, int hi)
+       {
+           System.Diagnostics.Trace.Assert(down_pif.host.opaque_ref == up_pif.host.opaque_ref);
+
+           int lo = action.PercentComplete;
+           int inc = (hi - lo) / 3;
+           lo += inc;
+           PoolManagementReconfigure_( action, up_pif, lo);
+
+           /* Pool recover slaves spwans two tasks dbsync (update_env) and server_init on each slaves. 
+            * Only after their completion master will be able to execute reconfigure_IPs. 
+            * Hence, we allow some setteling time for them to complete.
+            */
+           Thread.Sleep(10000);

```

Comment:
@stephen-turner We have tried a range of solutions like polling of tasks that gets created for dbsync and waited for a predefined time until they complete, but not always they were able to capture them. There is no other property as far as we know that we can check to ensure that emergency reset master has indeed completed in all slaves.